src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/Mutect2AlleleFilter.java

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect.filtering;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public abstract class Mutect2AlleleFilter<T> extends Mutect2Filter {
+
+    public Map<Allele, Double> applyFilter(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
+        LinkedHashMap<Allele, List<T>> dataByAllele = vc.getAlleles().stream().collect(Collectors.toMap(Function.identity(), allele -> new ArrayList<>(), (a, b) -> a, () -> new LinkedHashMap<>()));
+        LinkedHashMap<Allele, Double> probabilityByAllele = vc.getAlleles().stream().collect(Collectors.toMap(Function.identity(), allele -> this.getDefaultProbability(), (a, b) -> null, () -> new LinkedHashMap<>()));
+        vc.getGenotypes().stream().filter(filteringEngine::isTumor)
+                .filter(getPredicateForData())
+                .forEach(g -> {
+                    Iterator<T> alleleDataIterator = getDataForAlleles(g).iterator();
```

Comment:
`getDataForAlleles` is a `List`, rather than a `Map` because there is a fixed order of alleles.  This makes sense.  Why then isn't `dataByAllele` a `List<List<T>>` reflecting the same order?  That would then enable the following code to be written as
```
final List<T> sampleData = getDataForAlleles(g);
new IndexRange(0, numAlleles).forEach(a -> dataByAllele.get(a).add(sampleData.get(a)));
```
To my anti-C++ tastes that's clearer than iterators.

Reply:
when i was using ListOrderedMap instead of LinkedHashMap, I was able to do this:

`final int numItems = Math.min(dataByAllele.size(), alleleData.size());	                    Iterator<List<T>> dataByAlleleIterator = dataByAllele.values().iterator();
                    IntStream.range(0, numItems).forEach(i -> dataByAllele.getValue(i).add(alleleData.get(i)));`

I think it's important to associate the allele with the data when invoking the methods that will be overridden by the actual filter classes.

Reply:
That's a good argument for reverting to `ListOrderedMap`, in my opinion.